src/Domain/CoordinateMaps/CubicScale.cpp

```diff
@@ -101,168 +168,258 @@ boost::optional<std::array<tt::remove_cvref_wrap_t<T>, 1>> CubicScale::inverse(
+    // optimization for linear radial scaling
+    std::array<tt::remove_cvref_wrap_t<T>, Dim> result{};
+    for (size_t i = 0; i < Dim; ++i) {
+      gsl::at(result, i) = dt_a_of_t * gsl::at(source_coords, i);
+    }
+    return result;
+  }
+
   const auto dt_b_of_t = map_list.at(f_of_t_b_)->func_and_deriv(time)[1][0];
-  const auto frame_vel =
-      source_coords[0] *
-      (dt_a_of_t +
-       (dt_b_of_t - dt_a_of_t) * square(source_coords[0] / outer_boundary_));
 
-  return {{frame_vel}};
+  tt::remove_cvref_wrap_t<T> r_squared =
+      square(dereference_wrapper(source_coords[0]));
+  for (size_t i = 1; i < Dim; ++i) {
+    r_squared += square(dereference_wrapper(gsl::at(source_coords, i)));
+  }
+  // Reuse r^2 allocation
+  r_squared =
+      dt_a_of_t + (dt_b_of_t - dt_a_of_t) / square(outer_boundary_) * r_squared;
+
+  std::array<tt::remove_cvref_wrap_t<T>, Dim> result{};
+  for (size_t i = 0; i < Dim - 1; ++i) {
+    gsl::at(result, i) = gsl::at(source_coords, i) * r_squared;
+  }
+  r_squared = source_coords[Dim - 1] * r_squared;
+  result[Dim - 1] = std::move(r_squared);
+  return result;
 }
 
+template <size_t Dim>
 template <typename T>
-tnsr::Ij<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame> CubicScale::jacobian(
-    const std::array<T, 1>& source_coords, const double time,
+tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame>
+CubicScale<Dim>::jacobian(
+    const std::array<T, Dim>& source_coords, const double time,
     const std::unordered_map<
         std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
         map_list) const noexcept {
   const auto a_of_t = map_list.at(f_of_t_a_)->func(time)[0][0];
+
+  if (functions_of_time_equal_) {
+    // optimization for linear radial scaling
+    auto jac{make_with_value<
+        tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame>>(
+        dereference_wrapper(source_coords[0]), 0.0)};
+    for (size_t i = 0; i < Dim; ++i) {
+      jac.get(i, i) = a_of_t;
+    }
+    return jac;
+  }
+
   const auto b_of_t = map_list.at(f_of_t_b_)->func(time)[0][0];
-  auto jac{
-      make_with_value<tnsr::Ij<tt::remove_cvref_wrap_t<T>, 1, Frame::NoFrame>>(
-          dereference_wrapper(source_coords[0]), 0.0)};
 
-  get<0, 0>(jac) =
-      a_of_t +
-      3.0 * (b_of_t - a_of_t) * square(source_coords[0] / outer_boundary_);
+  tt::remove_cvref_wrap_t<T> r_squared =
+      square(dereference_wrapper(source_coords[0]));
+  for (size_t i = 1; i < Dim; ++i) {
+    r_squared += square(dereference_wrapper(gsl::at(source_coords, i)));
+  }
+  // Reuse r^2 allocation
+  r_squared = a_of_t + (b_of_t - a_of_t) / square(outer_boundary_) * r_squared;
+
+  const double coeff = 2.0 * (b_of_t - a_of_t) / square(outer_boundary_);
+
+  tnsr::Ij<tt::remove_cvref_wrap_t<T>, Dim, Frame::NoFrame> jac{};
+  for (size_t i = 0; i < Dim; ++i) {
+    for (size_t j = 0; j < Dim; ++j) {
+      if (i == j) {
+        jac.get(i, j) = r_squared + gsl::at(source_coords, i) *
```

Comment:
Would it make difference to just set to gsl::at(source_coords, i) * gsl::at(source_coords, j) * coeff and then add r_squared if i == j, with no "else"? It's a minor nit to pick, but I would have thought that would be better, for minimizing a bit of duplicated code

Reply:
The reason I wrote it this way is so we only need to loop through the DataVector once

Reply:
Again, not a big deal, just trying to understand. I think I get it now: my way would have 3 ifs instead of one if/else.